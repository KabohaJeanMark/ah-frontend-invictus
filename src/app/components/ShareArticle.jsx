import React from 'react';
import { Helmet } from "react-helmet";
import { APP_URLS } from '../../utilities';
import '../../assets/styles/shareArticleButtons.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEnvelopeSquare } from '@fortawesome/free-solid-svg-icons';
import { faFacebookSquare, faTwitterSquare } from '@fortawesome/free-brands-svg-icons';

export const socialRedirect = (url) => {
  window.open(url, '_blank');
};

export default ({ title, description, social_links: { facebook, twitter, email } }) => {
  const datas = [
    { media: facebook, icon: faFacebookSquare, share: "Facebook" },
    { media: twitter, icon: faTwitterSquare, share: "Twitter" },
    { media: email, icon: faEnvelopeSquare, share: "Email" }
  ];
  const links = datas.map(data => (
    <div className="col-sm-4 icon toolstip">
      <FontAwesomeIcon onClick={() => socialRedirect(data.media)} icon={data.icon} />
      <span className="toolstiptext">Share to {data.share}</span>
    </div>
  ));
  return (
    <div className="shareicons float-right">
      <Helmet>
        <title>{title}</title>
        <url>{APP_URLS.FRONT_END}</url>
        <type>article</type>
        <description>{description}</description>
        <image>https://picsum.photos/600/300</image>
        <meta property="og:url" content={APP_URLS.FRONT_END} />
        <meta property="og:type" content="article" />
        <meta property="og:title" content={title} />
        <meta property="og:description" content={description} />
        <meta property="og:image" content="https://picsum.photos/600/300" />
      </Helmet>
      <div className="row">
        {links}
      </div>
    </div>
  );
};
