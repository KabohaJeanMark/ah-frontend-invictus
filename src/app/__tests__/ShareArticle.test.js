import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ShareArticle, { socialRedirect } from '../components/ShareArticle';

const props = {
  title: "title",
  description: "description",
  social_links: {
    facebook: 'url',
    twitter: 'url',
    email: 'url'
  }};

describe('article component', () => {
  let wrapper = shallow(<ShareArticle {...props} />);
  socialRedirect('url');

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
