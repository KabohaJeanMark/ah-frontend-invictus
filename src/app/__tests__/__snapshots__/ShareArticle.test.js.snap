// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article component matches snapshot 1`] = `
<div
  className="shareicons float-right"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      title
    </title>
    <url>
      https://staging-f-invictus.herokuapp.com
    </url>
    <type>
      article
    </type>
    <description>
      description
    </description>
    <image>
      https://picsum.photos/600/300
    </image>
    <meta
      content="https://staging-f-invictus.herokuapp.com"
      property="og:url"
    />
    <meta
      content="article"
      property="og:type"
    />
    <meta
      content="title"
      property="og:title"
    />
    <meta
      content="description"
      property="og:description"
    />
    <meta
      content="https://picsum.photos/600/300"
      property="og:image"
    />
  </HelmetWrapper>
  <div
    className="row"
  >
    <div
      className="col-sm-4 icon toolstip"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f082",
              "M400 32H48A48 48 0 0 0 0 80v352a48 48 0 0 0 48 48h137.25V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.27c-30.81 0-40.42 19.12-40.42 38.73V256h68.78l-11 71.69h-57.78V480H400a48 48 0 0 0 48-48V80a48 48 0 0 0-48-48z",
            ],
            "iconName": "facebook-square",
            "prefix": "fab",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        onClick={[Function]}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <span
        className="toolstiptext"
      >
        Share to 
        Facebook
      </span>
    </div>
    <div
      className="col-sm-4 icon toolstip"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f081",
              "M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-48.9 158.8c.2 2.8.2 5.7.2 8.5 0 86.7-66 186.6-186.6 186.6-37.2 0-71.7-10.8-100.7-29.4 5.3.6 10.4.8 15.8.8 30.7 0 58.9-10.4 81.4-28-28.8-.6-53-19.5-61.3-45.5 10.1 1.5 19.2 1.5 29.6-1.2-30-6.1-52.5-32.5-52.5-64.4v-.8c8.7 4.9 18.9 7.9 29.6 8.3a65.447 65.447 0 0 1-29.2-54.6c0-12.2 3.2-23.4 8.9-33.1 32.3 39.8 80.8 65.8 135.2 68.6-9.3-44.5 24-80.6 64-80.6 18.9 0 35.9 7.9 47.9 20.7 14.8-2.8 29-8.3 41.6-15.8-4.9 15.2-15.2 28-28.8 36.1 13.2-1.4 26-5.1 37.8-10.2-8.9 13.1-20.1 24.7-32.9 34z",
            ],
            "iconName": "twitter-square",
            "prefix": "fab",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        onClick={[Function]}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <span
        className="toolstiptext"
      >
        Share to 
        Twitter
      </span>
    </div>
    <div
      className="col-sm-4 icon toolstip"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f199",
              "M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zM178.117 262.104C87.429 196.287 88.353 196.121 64 177.167V152c0-13.255 10.745-24 24-24h272c13.255 0 24 10.745 24 24v25.167c-24.371 18.969-23.434 19.124-114.117 84.938-10.5 7.655-31.392 26.12-45.883 25.894-14.503.218-35.367-18.227-45.883-25.895zM384 217.775V360c0 13.255-10.745 24-24 24H88c-13.255 0-24-10.745-24-24V217.775c13.958 10.794 33.329 25.236 95.303 70.214 14.162 10.341 37.975 32.145 64.694 32.01 26.887.134 51.037-22.041 64.72-32.025 61.958-44.965 81.325-59.406 95.283-70.199z",
            ],
            "iconName": "envelope-square",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        onClick={[Function]}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <span
        className="toolstiptext"
      >
        Share to 
        Email
      </span>
    </div>
  </div>
</div>
`;
